blockstack/blockstack-browser
# Attempt to Connect Storage on Load

### What This Does

Closes #1562. Closes #1537. This is a theoretical fix for the previous two issues. The problem is that when the app gets in a state where storage hasn't been initialized (Checked with `state.settings.api.storageConnected`) that the whole app gets screwy. There was clearly _some_ consideration for this state, because a lot of code paths early exit when this happens. But there's no communication to the user about what's going on, and things just silently don't work or are disabled.

What this PR does is, similar to checking if you've signed in or created an account, it also checks if your storage is connected. If it's not, it pops you over to a `/connect-storage` page that shows a loader while the storage is config'd, then redirects you back to where you were going. It often goes so fast it's pretty imperceptible, but it should fix up their storage woes.

### Steps to Test

1. Get your app into a state of `!storageConnected`
    * You can either do this by following the steps in #1537, or by manually editing that field in your local storage JSON blob
2. Attempt to load a regular route (e.g. `/`, or try to auth an app.)
3. Confirm you were sent to `/connect-storage` and that it fixed your storage
4. Confirm you were then redirected to your original path

---

As an aside, this also cleans up some copy-paste code. I noticed we're not often using the `getState` param from redux thunks, but this code should demonstrate that you can make components _much simpler_ by having actions coordinate complex interactions. I'd like to see us use this pattern more.
